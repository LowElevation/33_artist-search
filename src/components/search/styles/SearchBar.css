* {
    margin: 0;
    padding: 0;
}

.SearchBar {
    display: flex;
    align-items: center;
    justify-content: center;

    button, input {
        height: 30px;
        margin: 5px;
    }

    button {
        width: 40px;
    }
}