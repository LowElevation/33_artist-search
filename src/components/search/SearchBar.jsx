import React from "react";
import PropTypes from "prop-types";
import styles from "./styles/SearchBar.css";

const SearchBar = ({ search, handleSearch }) => {
  return (
    <form data-testid="Go" className={styles.SearchBar} onSubmit={handleSearch}>
      <input defaultValue={search} />
      <button>Go</button>
    </form>
  );
};

SearchBar.propTypes = {
  search: PropTypes.string.isRequired,
  handleSearch: PropTypes.func.isRequired,
  
};

export default SearchBar;
