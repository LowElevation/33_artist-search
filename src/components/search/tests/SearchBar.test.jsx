import React from "react";
import { MemoryRouter as Router } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import SearchBar from "../SearchBar";

describe("SearchBar component", () => {
  afterEach(() => cleanup());
  it("renders SearchBar", () => {
    const { asFragment } = render(
      <Router>
        <SearchBar search={""} handleSearch={jest.fn()} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
