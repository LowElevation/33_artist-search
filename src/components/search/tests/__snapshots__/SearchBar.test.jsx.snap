// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders SearchBar 1`] = `
<DocumentFragment>
  <form
    class="SearchBar"
  >
    <input
      value=""
    />
    <button>
      Go
    </button>
  </form>
</DocumentFragment>
`;
